app/cdap/components/CreateRecipe/index.tsx

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { FormEvent, ChangeEvent, useEffect, useState, useRef } from 'react';
+import T from 'i18n-react';
+import { getRecipeByName, createRecipe } from 'components/CreateRecipe/services';
+import { debounce } from 'lodash';
+import { IRecipeFormData } from 'components/RecipeManagement/types';
+import RecipeForm, { CREATE_RECIPE } from 'components/RecipeManagement/RecipeForm';
+import { ICreateRecipeProps } from 'components/RecipeManagement/types';
+
+export default function CreateRecipe({ setShowRecipeForm, setSnackbar }: ICreateRecipeProps) {
+  const [recipeFormData, setRecipeFormData] = useState<IRecipeFormData>({
+    recipeName: '',
+    description: '',
+    directives: [],
+  });
+
+  const [recipeNameErrorMessage, setRecipeNameErrorMessage] = useState<string>('');
+  const [isRecipeNameError, setIsRecipeNameError] = useState<boolean>(false);
```

Comment:
I see that both states `recipeNameErrorMessage` and `isRecipeNameError` serves the same purpose. Can't we use one state instead of maintaining two states?  
or Is there any case where  `isRecipeNameError` would be true without `recipeNameErrorMessage` ? 
Please check on this.

Reply:
Fixed by using one state